### ISSUE TITLE
Formula evaluation issue

### ISSUE THREAD

**Issue by Ingenu1ty** [saltcorn/saltcorn#2632]
I use the following formula in a text box (I assume formulas elsewhere have the same problems):
`k_to_user.fnln[0] + ". " + (k_to_user.fnln).match(/([ ][^ ])(?=[^ ])/g)[0]`

and I get an error: 
`Unable to build a joind without a reftable_name.`

It seems that the `k_to_user.fnln` does not resolve to a string if it is followed by a method call(dot notation)

**Comment by Ingenu1ty** [comment]
I think I've found a workaround. Quite an ugly workaround :(


``` javascript 
`${k_to_user.fnln[0]}.  ${(`${k_to_user.fnln}`).match(/([ ][^ ])(?=[^ ])/g)[1]}` 
```

At least this can help those who have encountered the same problem. And of course it **would be better** if everything worked as expected.

**Comment by glutamate** [comment]
@Ingenu1ty can you just clarify in your example what is `fnln` - is this the name of a field on a related table? and if not where does it come from. What are the field types on the one/two tables

**Comment by Ingenu1ty** [comment]
fnln is first name, last name like "John Connor" :) I just want to display "J.C." in the text label

> what is fnln - is this the name of a field on a related table?

yes exactly👍 There is a view on the, let's say, "schedule" table. `schedule` has a field type "key to users"(k_to_user). And in the table "users" there is a field fnln of type string

**Comment by glutamate** [comment]
fixed in #2702
